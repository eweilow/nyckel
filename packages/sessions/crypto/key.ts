import crypto from "crypto";
import { saltKey } from "./salt";

export function createKeyHash(id: string, salt: string): string {
  const cipher = crypto.createHash("sha256");
  cipher.update(saltKey(id, salt), "utf8");
  return cipher.digest("base64");
}
