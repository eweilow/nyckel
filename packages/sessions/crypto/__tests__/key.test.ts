import { createKeyHash } from "../key";

describe("createKeyHash", () => {
  it("returns a hash", () => {
    expect(createKeyHash("id", "salt")).toMatchSnapshot();
  });
  it("returns the same thing for given id and salt", () => {
    expect(createKeyHash("id", "salt")).toBe(createKeyHash("id", "salt"));
  });
});
