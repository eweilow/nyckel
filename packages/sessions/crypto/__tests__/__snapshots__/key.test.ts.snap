// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createKeyHash returns a hash 1`] = `"6k/PzHQ3nixDDJ+oo+/tuGzYlv9tZmBFIXB3kj3053o="`;
